core/src/main/java/org/lflang/target/Target.java

```diff
@@ -546,4 +580,84 @@ public static Target fromDecl(TargetDecl targetDecl) {
     return Target.forName(name)
         .orElseThrow(() -> new RuntimeException("Invalid target name '" + name + "'"));
   }
+
+  public void initialize(TargetConfig config) {
```

Comment:
This still seems a bit backwards. Instead of having a switch case here, why not add a method to `GeneratorBase` that is extended by the individual target generators and to the registering their. Then the generators would be responsible for declaring the properties that they support.

Reply:
OK, that works for me. Will make the change.

Reply:
Actually, I don't think this will work. The problem is that the target configuration is part of the context, which is passed into the generator and is also used during validation. We need to register the properties ahead of time, or else we cannot catch during validation whether a property is supported or not.

Reply:
So, whether this is backwards or not, let's keep it like this for now and prioritize getting this merged. So far, we only have a single code generator for each target, so it shouldn't really matter where we do the initialization.